sched/environ/env_dup.c

```diff
@@ -61,14 +63,13 @@
  *
  ****************************************************************************/
 
-int env_dup(FAR struct task_group_s *group)
+int env_dup(FAR struct task_group_s *group, FAR char * const *envp,
```

Comment:
This is still a bit ugly, it should be possible to calculate the length of the environment from envp but it is not possible due to how NuttX currently stores and provisions the environment variables. The envp parameter is not really a "list of pointers-to-string", it is a buffer that contains several appended strings. Thus, the size of envp cannot be calculated iteratively as there is no terminating element and adding one makes the implementation even more complex.

The real solution would be to store envp as list of pointers-to-strings, like the canonical implementation does, but this is too much work and I don't even need env for anything, I just want to fix the crashing when the env string is copied from parent to child.

Reply:
It can be fixed to let envp terminated by the double "\0\0", which's a common approach to indicate the end of a string array.

Reply:
Ok.

Reply:
This is a bugfix; the original bug causes a full system crash and I need this as well.

I think it is reasonable to integrate the bugfix and do the refactoring later, if needed.

Reply:
The current design principle is that the environment is stored in the tcb.group structure and that is copied to the child process. This commit just tries to fix an issue with the old implementation.

I fully agree the storage and provisioning of envp could be improved upon, specifically handle it like argv and pass it through from the POSIX standard functions like posix_spawn(); But the current implementation works, needs only this 1 fix to work for CONFIG_BUILD_KERNEL, and it can always be refactored later.

Reply:
@pussuw https://github.com/apache/incubator-nuttx/pull/6064 ensure envp terminated by double '\0'.